metadata-integration/java/spark-lineage/src/main/java/datahub/spark/DatasetExtractor.java

```diff
@@ -87,45 +84,43 @@
       }
 
       String tbl = options.get("dbtable");
-      return Optional.of(new JdbcDataset(url, tbl));
+      return Optional.of(new JdbcDataset(url, tbl, getCommonPlatformInstance(datahubConfig), getCommonFabricType(datahubConfig)));
     });
 
-    PLAN_TO_DATASET.put(CreateDataSourceTableAsSelectCommand.class, (p, ctx) -> {
+    PLAN_TO_DATASET.put(CreateDataSourceTableAsSelectCommand.class, (p, ctx, datahubConfig) -> {
       CreateDataSourceTableAsSelectCommand cmd = (CreateDataSourceTableAsSelectCommand) p;
       // TODO what of cmd.mode()
-      return Optional.of(new CatalogTableDataset(cmd.table()));
+      return Optional.of(new CatalogTableDataset(cmd.table(), getCommonPlatformInstance(datahubConfig), getCommonFabricType(datahubConfig)));
```

Comment:
is there a good reason to compute the fabricType and platformInstance everytime, versus just calculating it once and using a final variable in the other method calls?

Reply:
We need to extend this further to each dataset level platforminstance. So there will be current dataset specific selection. Thats why i have not get it done at common place